From 0de80b5c02ad3b6e987c84af82bfb1d506157749 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 28 Nov 2014 13:52:46 +0800
Subject: [PATCH 159/164] atomisp2: Check pipe id to identify post view when
 re-triggering capture

In SDV mode, the preview buffers are also from the VF output of video pipe.
Need to chek pipe id when identify the post view buffer.

Change-Id: Iac7aabff678cca9bc56c9b99c1719f04a716c80f
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-4029
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/30524
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_cmd.c      |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 910c2d4..ea72a74 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1049,7 +1049,8 @@ void atomisp_buf_done(struct atomisp_sub_device *asd, int error,
 			WARN_ON(!vb);
 			if (vb)
 				pipe->frame_config_id[vb->i] = frame->isp_config_id;
-			if (asd->pending_capture_request > 0) {
+			if (css_pipe_id == IA_CSS_PIPE_ID_CAPTURE &&
+			    asd->pending_capture_request > 0) {
 				err = atomisp_css_offline_capture_configure(
 					asd,
 					asd->params.offline_parm.num_captures,
-- 
1.7.9.5

